src/Stoa.ts

```diff
@@ -173,8 +173,17 @@ class Stoa extends WebService
                     }
                     else
                     {
-                        preimage_distance = NaN;
-                        result_preimage_hash = Hash.NULL;
+                        // Genesis block height
+                        if (target_height == new Height(0n))
+                        {
+                            preimage_distance = 0;
+                            result_preimage_hash = row.random_seed;
+                        }
+                        else
+                        {
+                            preimage_distance = NaN;
+                            result_preimage_hash = Hash.NULL;
+                        }
```

Comment:
Why is it special-casing the genesis block ? Any enrollment contains the preimage at distance 0.

Reply:
That's right. I'm testing it now.

Reply:
I fixed it